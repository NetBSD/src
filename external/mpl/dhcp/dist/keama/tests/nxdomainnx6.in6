# nxdomain numeric expression

# empty configs are not accepted by Kea
default-lease-time 1800;

# reduce literals
class "literal" {
    match if substring(option dhcp6.interface-id, 0, nxdomain % 128) = ab:cd;
}

# raw
option dhcp6.interface-id = encode-int(nxdomain + bound, 32);
