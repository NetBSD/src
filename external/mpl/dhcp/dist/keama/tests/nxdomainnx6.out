{
  # nxdomain numeric expression
  # empty configs are not accepted by Kea
  "Dhcp6": {
    "valid-lifetime": 1800,
    "client-classes": [
      # reduce literals
      {
        "name": "literal",
        /// from: match if (substring(option dhcp6.interface-id, 0, 393231 % 128)) = 0xabcd
        "test": "substring(option[18].hex,0,15) == 0xabcd"
      }
    ],
    "option-data": [
      # raw
      {
        "space": "dhcp6",
        "name": "interface-id",
        "code": 18,
        "csv-format": false,
//      /// constant DHCP_R_NXDOMAIN(393231)
//      /// constant S_BOUND(5)
//      "original-data": "\u0000\u0006\u0000\u0014",
        "data": "00060014"
      }
    ]
  }
}
